code/modules/vehicles/spacepod/_pod.dm

```diff
@@ -0,0 +1,280 @@
+					occupant.clear_alert(ALERT_CHARGE)
+				if(0.5 to 0.75)
+					occupant.throw_alert(ALERT_CHARGE, /atom/movable/screen/alert/lowcell, 1)
+				if(0.25 to 0.5)
+					occupant.throw_alert(ALERT_CHARGE, /atom/movable/screen/alert/lowcell, 2)
+				if(0.01 to 0.25)
+					occupant.throw_alert(ALERT_CHARGE, /atom/movable/screen/alert/lowcell, 3)
+				else
+					occupant.throw_alert(ALERT_CHARGE, /atom/movable/screen/alert/emptycell)
+		else
+			occupant.throw_alert(ALERT_CHARGE, /atom/movable/screen/alert/nocell)
+		var/integrity = get_integrity_percentage() * 100
+		switch(integrity)
+			if(40 to 60)
+				occupant.throw_alert(ALERT_MECH_DAMAGE, /atom/movable/screen/alert/pod_damage, 1)
+			if(20 to 40)
+				occupant.throw_alert(ALERT_MECH_DAMAGE, /atom/movable/screen/alert/pod_damage, 2)
+			if(-INFINITY to 20)
+				occupant.throw_alert(ALERT_MECH_DAMAGE, /atom/movable/screen/alert/pod_damage, 3)
+			else
+				occupant.clear_alert(ALERT_MECH_DAMAGE)
+
+// handles giving equipment actions + some other stuff
+/obj/vehicle/sealed/space_pod/after_add_occupant(mob/living/occupant)
+	. = ..()
+	occupant.refresh_gravity()
+	if(length(occupants) == 1) //first occupant only
+		panel_open = FALSE //automatic screws,,,, waow....
+		cycle_tank_air()
+	for(var/obj/item/pod_equipment/equipment as anything in get_all_parts())
+		var/datum/action/action = equipment.create_occupant_actions(occupant, occupants[occupant])
+		if(isnull(action))
+			continue
+		if(islist(action))
+			var/list/as_list = action
+			for(var/datum/action/actual_action as anything in as_list)
+				actual_action.Grant(occupant)
+		else
+			action.Grant(occupant)
+		equipment_actions[occupant] += islist(action) ? action : list(action)
+
+//removes equipment actions
+/obj/vehicle/sealed/space_pod/after_remove_occupant(mob/living/former)
+	. = ..()
+	former.refresh_gravity()
+	former.clear_alert(ALERT_CHARGE)
+	former.clear_alert(ALERT_MECH_DAMAGE)
+	if(!length(occupants)) //when everyone exits
+		cycle_tank_air(to_tank = TRUE)
+	if(equipment_actions[former])
+		QDEL_LIST(equipment_actions[former])
+		equipment_actions -= former
+
+// atmos
+/obj/vehicle/sealed/space_pod/proc/cycle_tank_air(to_tank = FALSE)
+	if(isnull(cabin_air_tank))
+		return
+	var/datum/gas_mixture/from = to_tank ? cabin_air : cabin_air_tank.return_air()
+	var/datum/gas_mixture/target = to_tank ? cabin_air_tank.return_air() : cabin_air
+	var/datum/gas_mixture/removed = from.remove(from.total_moles())
+	if(!removed)
+		return
+	target.merge(removed)
+
+/obj/vehicle/sealed/space_pod/remove_air(amount)
+	return !isnull(cabin_air_tank) ? cabin_air.remove(amount) : ..()
+/obj/vehicle/sealed/space_pod/return_air()
+	return !isnull(cabin_air_tank) ? cabin_air : ..()
+/obj/vehicle/sealed/space_pod/return_analyzable_air()
+	return !isnull(cabin_air_tank) ? cabin_air : null // no internal air
+/obj/vehicle/sealed/space_pod/return_temperature()
+	var/datum/gas_mixture/air = return_air()
+	return air?.return_temperature()
+
```

Comment:
newlines please, also is this mecha copypasta? should we have a type in between them and pods? or something else

Reply:
we cant make a type between them because both have their own atmos behavior